Develop

N1khto opened the pull request:
DB structure:
![Pet service DB structure](https://github.com/N1khto/pet-owner-service/assets/112682735/eab16fb1-5f04-4845-aebb-de5ced917eba)

Screenshots:
![Знімок екрана (1)](https://github.com/N1khto/pet-owner-service/assets/112682735/5af7412a-045a-4d79-9c3e-7849402d1e04)
![Знімок екрана (2)](https://github.com/N1khto/pet-owner-service/assets/112682735/c065637e-817f-47b5-a313-d3055f448236)
![Знімок екрана (3)](https://github.com/N1khto/pet-owner-service/assets/112682735/d9701615-6c91-4382-9845-8fb2b081ba82)
![Знімок екрана (4)](https://github.com/N1khto/pet-owner-service/assets/112682735/55ca6dad-cf73-424c-b255-24fd652d7601)
![Знімок екрана (5)](https://github.com/N1khto/pet-owner-service/assets/112682735/21a59926-b8d6-4274-ba56-64d276d1e4b1)
![Знімок екрана (6)](https://github.com/N1khto/pet-owner-service/assets/112682735/361451e0-d903-4012-a7ed-c719b5474fb4)
![Знімок екрана (7)](https://github.com/N1khto/pet-owner-service/assets/112682735/eb43c4da-6862-4ce0-a9a7-7584768335fe)
![Знімок екрана (8)](https://github.com/N1khto/pet-owner-service/assets/112682735/42f057ff-4802-41ee-8d61-d0e842b4b856)
![Знімок екрана (9)](https://github.com/N1khto/pet-owner-service/assets/112682735/b4a3732b-099b-4716-a7cf-13b4d5314e6e)
![Знімок екрана (10)](https://github.com/N1khto/pet-owner-service/assets/112682735/278767a0-ff58-479c-8cdc-40b11271c64d)
![Знімок екрана (11)](https://github.com/N1khto/pet-owner-service/assets/112682735/ae4454dc-f7c6-4538-b004-4a024ce00b55)
![Знімок екрана (12)](https://github.com/N1khto/pet-owner-service/assets/112682735/031f2a02-a04d-4de3-8691-c9bb2b4e9b46)
![Знімок екрана (13)](https://github.com/N1khto/pet-owner-service/assets/112682735/7661f624-a6b9-4ee4-9cb1-6930e53b287e)
![Знімок екрана (14)](https://github.com/N1khto/pet-owner-service/assets/112682735/8891c2fc-7769-458c-b278-81d4194e4d92)
![Знімок екрана (15)](https://github.com/N1khto/pet-owner-service/assets/112682735/bf29ea69-41c5-4416-997b-c30e1830f0e8)
![Знімок екрана (16)](https://github.com/N1khto/pet-owner-service/assets/112682735/72afbf08-bc65-4a69-b66e-59934e3ef9ef)
![Знімок екрана (17)](https://github.com/N1khto/pet-owner-service/assets/112682735/b3dc3593-8992-4865-b3e4-ae0d5dc06458)

N1khto committed:
added empty django project pet_service and empty app pet_app

N1khto committed:
Created 5 models: Pet, Species, PetFood, Brand and PetOwner. Last set as user model of the pet_app. And made firs db migration.

N1khto committed:
Created 5 models: Pet, Species, PetFood, Brand and PetOwner. Last set as user model of the pet_app. And made firs db migration.

N1khto committed:
1. Updated models:
 - PetOwner got class Meta with verbose_name and verbose_name_plural
 - Species, Brand, PetFood models got new __str__ methods.
 2. Registered all the models in admin.py
 3. Three new db migrations related to changes

N1khto committed:
1. Created file urls.py in pet_app directory and included it in project urls.py.
2. Added path to homepage in pet_app/urls.py
3. Created index view for homepage in views.py
4. Created dir templates and register it in settings
5. Created empty html template for index page

N1khto committed:
Added static dir and needed settings

N1khto committed:
1. Added bootstrap theme with static files
2. Created base.html file and implemented raw homepage(index.html) of the pet service
3. Separated footef form base.html and save it in templates\includes

N1khto committed:
1. Moved navigation and sidebar form base template to separate files into includes dir.
2. Added pagination includes and custom tag gor it.

N1khto committed:
Implemented species list view and template

N1khto committed:
1. Created forms.py file with search form for species list page
2. Added search form template to includes
3. Added crispy forms to installed apps
4. Changed species list view to interfare with search form

N1khto committed:
1. Added login\logout templates and corresponding path to urls
2. Edited sidebar template

N1khto committed:
Added create\update\delete templates and views for species

N1khto committed:
Added list\detail\create\update\delete templates and views for pet model

N1khto committed:
Added list\create\update\delete templates and views for brand model

N1khto committed:
Added external links on top navbar in base template

N1khto committed:
Added list\detail\create\update\delete templates and views for pet_owner model

N1khto committed:
Added list\detail\create\update\delete templates and views for pet_food model

N1khto committed:
1. Added background image to base template
2. Edited species list template
3. Disabled footer(could be deleted in future)

N1khto committed:
Edited appearance of pet, pet_food, brand and pet_owner list templates

N1khto committed:
Added clickable links in brand, pet and pet_food list templates

N1khto committed:
Installed django debug toolbar

N1khto committed:
Fixed N+1 problem(modified querysets) in Pet, PetOwner and Brand list views

N1khto committed:
1. Edited index page
2. Edited pet, pet owner and pet food detail templates - added information and links

N1khto committed:
Fixed N+1 problem in pet detail view - edited queryset

N1khto committed:
Added validation of pet_owner_experience attr in pet owner create and update forms(limit 100 years)

N1khto committed:
1. Deleted footer
2. Edited readme.md - added info and test user credentials.

N1khto committed:
1.Save Mate icon to static dir and change base.html to use it from there
2. Edit multiple python code files to keep up with flake8 tests
3. Formatted html templates in includes dir with builtin autoformat.

N1khto committed:
Formatted hase.html and all html templates in pet_app and registration dir with builtin autoformatting.

ArsenPidhoretskyi commented on pet_app/models.py:
```diff
@@ -0,0 +1,65 @@
+
+
+class PetOwner(AbstractUser):
+    pet_owner_experience = models.IntegerField(default=0)
```
FYI better to use PositiveIntegerField

ArsenPidhoretskyi commented on pet_app/models.py:
```diff
@@ -0,0 +1,65 @@
+    nickname = models.CharField(max_length=255)
+    age = models.IntegerField()
+    gender = models.CharField(max_length=15, choices=PET_GENDER, null=True)
+    owner = models.ForeignKey(get_user_model(), on_delete=models.CASCADE)
+    species = models.ForeignKey(Species, on_delete=models.SET_NULL, null=True)
```
Provide related names

N1khto committed:
Fixes:
1.Edited PetOwner model - changed IntegerField in pet_owner_experience attr to PositiveIntegerField.
2. Provided related names for species and owner attrs in Pet model
3. Edited pet_owner_detail and pet_owner_list templates to keep up with changes
